libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -551,6 +585,117 @@ impl NodeAPI for Greenlight {
         })
     }
 
+    async fn send_pay(&self, bolt11: String, max_hops: u32) -> NodeResult<PaymentResponse> {
+        let invoice = parse_invoice(&bolt11)?;
+        let last_hop = invoice.routing_hints.first().and_then(|rh| rh.hops.first());
+        let mut client: node::ClnClient = self.get_node_client().await?;
+
+        // We first calculate for each channel the max amount to pay (at the receiver)
+        let (mut max_amount_per_channel, payment_path) = self
+            .max_amount_to_send(Some(hex::decode(invoice.payee_pubkey)?), max_hops, last_hop)
+            .await?;
+        info!("send_pay: route: {:?}", payment_path);
+
+        // Calculate the total amount to pay
+        let total: u64 = max_amount_per_channel.iter().map(|m| m.amount_msat).sum();
+
+        // Sort the channels by max amount descending so we can build the route in a way that it
+        // drains the largest channels first
+        max_amount_per_channel.sort_by(|m1, m2| m2.amount_msat.cmp(&m1.amount_msat));
+
+        let amount_to_pay = match invoice.amount_msat {
+            Some(amount) => Ok(amount),
+            None => Err(NodeError::Generic(anyhow!("Invoice has no amount"))),
+        }?;
+
+        if amount_to_pay > total {
+            return Err(NodeError::RouteNotFound(anyhow!(
+                "amount too high, max amount is {}",
+                total
+            )));
+        }
+
+        // This is needed in greenlight for the signer to recognize this invoice.
+        client
+            .pre_approve_invoice(pb::cln::PreapproveinvoiceRequest {
+                bolt11: Some(bolt11.clone()),
+            })
+            .await?;
+
+        // We need to allocate a part id for each part that we are sending.
+        let mut part_id = 1;
+        // The total amount we sent
+        let mut amount_sent = 0;
+        // The total amount received
+        let mut amount_received = 0;
```

Comment:
Make it a little clearer what these variables mean.

```suggestion
        // The total amount we sent. i.e. what the recipient received + fees
        let mut total_amount_sent = 0;
        // The total amount received by the recipient
        let mut total_amount_received = 0;
```

Reply:
IMO the variable names should include the unit to avoid confusion.
(e.g. `total_amount_sent` -> `total_amount_sent_msat`)

Reply:
Fixed